src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,208 @@
+        }
+        return externalList;
+    }
+
+    private void getMeadicationsBetweenDatesFromDatabase(String initialDate, String finalDate,
+                                                         CollectionReference medsRef,
+                                                         List<Map<List<String>, Object>> externalList)
+            throws InterruptedException, ExecutionException {
+        String date;
+        List<String> aux = null;
+        ApiFuture<QuerySnapshot> future = medsRef.get();
+        List<QueryDocumentSnapshot> medDocuments = future.get().getDocuments();
+        for (QueryDocumentSnapshot medDocument : medDocuments) {
+            aux = pkToList(medDocument.getId());
+            date = pkToDate(medDocument.getId());
+            if (initialDate.compareTo(date) < 0 && finalDate.compareTo(date) > 0) {
+                Map<List<String>, Object> internalList = new HashMap<>();
+                internalList.put(Collections.singletonList(DATENAME), aux);
+                internalList.put(Collections.singletonList(BODY),
+                        medDocument.toObject(MedicationEntity.class));
+                externalList.add(internalList);
+            }
+            aux.clear();
+        }
+    }
+
+
+
+    @Override
+    public Object getMedicationField(String owner, String petName, String dateName, String field)
+            throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        DocumentReference medicationDocRef = medicationsRef.document(dateName);
+        ApiFuture<DocumentSnapshot> future = medicationDocRef.get();
+        DocumentSnapshot medicationDoc;
+        try {
+            medicationDoc = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        if (!medicationDoc.exists()) {
+            throw new DatabaseAccessException(INVALID_MEDICATION_EXC,
+                    MEDICATION_DOES_NOT_EXIST_EXC);
+        }
+        return medicationDoc.get(field);
+    }
+
+    @Override
+    public void updateMedicationField(String owner, String petName, String dateName, String field,
+                                      Object value) {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        DocumentReference medRef = medicationsRef.document(dateName);
+        medRef.update(field, value);
+    }
+
+
+    public CollectionReference getMedicationsRef(String owner, String petName) {
+        return db.collection("users").document(owner).collection("pets").document(petName).collection("medications");
+    }
+
+    public String pkToDate(String pk) {
+        String[] parts = pk.split(SEPARATOR);
+        return parts[0];
+    }
+
+    public String pkToName(String pk) {
+        String[] parts = pk.split(SEPARATOR);
+        return parts[1];
+    }
+
+    public List<String> pkToList(String pk){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)